code/modules/mob/living/death.dm

```diff
@@ -91,7 +91,6 @@
 		if(SSlag_switch.measures[DISABLE_DEAD_KEYLOOP] && !client?.holder)
 			to_chat(src, span_deadsay(span_big("Observer freelook is disabled.\nPlease use Orbit, Teleport, and Jump to look around.")))
 			ghostize(TRUE)
-	set_disgust(0)
```

Comment:
The disgust status effect should register `COMSIG_LIVING_DEATH` to `qdel(src)` when they die

Somewhat unrelated but it's strange that we allow effects to add disgust to dead mobs when we remove it all on death. Maybe we should also prevent adding disgust if `stat == DEAD`

Reply:
Yeah sure, I'll do both of these.